pysurfex/bufr.py

```diff
@@ -66,6 +65,9 @@ def __init__(
 
         # open bufr file
         file_handler = open(bufrfile, mode="rb")
+        lines_in_file = file_handler.readlines()
+        number_of_lines = int(len(lines_in_file))
```

Comment:
Looks a bit strange to see `readlines` for a file opened in the binary mode. Why not using a combination of `seek` and `tell` to get the file size and then `tell` in the loop to derive the percentage of processed data?